Core/Core/View/Base/Webview/WebView.swift

```diff
@@ -158,20 +145,23 @@ public struct WebView: UIViewRepresentable {
             }
             
             let baseURL = await parent.viewModel.baseURL
-            if !baseURL.isEmpty, !url.absoluteString.starts(with: baseURL) {
-                if navigationAction.navigationType == .other {
-                    return .allow
-                } else if navigationAction.navigationType == .linkActivated {
-                    await MainActor.run {
-                        UIApplication.shared.open(url, options: [:])
-                    }
-                } else if navigationAction.navigationType == .formSubmitted {
-                    return .allow
+            switch navigationAction.navigationType {
+            case .other:
+                return .allow
+            case .linkActivated:
+                await MainActor.run {
+                    UIApplication.shared.open(url, options: [:])
                 }
                 return .cancel
+            case .formSubmitted:
+                return .allow
+            default:
+                if !baseURL.isEmpty, !url.absoluteString.starts(with: baseURL) {
+                    return .cancel
+                } else {
+                    return .allow
+                }
```

Comment:
we can have only one case, `.linkActivated` and all the other cases can be handled in `default:`

Reply:
by some reason types `.other` and `.formSubmitted` was allowed and all other was cancelled `if !baseURL.isEmpty, !url.absoluteString.starts(with: baseURL)`
do you have some idea why?
I left the same logic as before